Bug: MdSidenav [opened] causes unresolved event _openPromiseReject

Bug when using MdSidenav open as default:
`<md-sidenav #demo opened="true">`

The browser claims of an error when trying to close it with 
`<button md-button (click)="demo.close() / toggle()">Close</button>`

```
core.umd.js:3427 EXCEPTION: Uncaught (in promise): undefinedErrorHandler.handleError @ core.umd.js:3427next @ core.umd.js:6901schedulerFn @ core.umd.js:6150SafeSubscriber.__tryOrUnsub @ Subscriber.ts:238SafeSubscriber.next @ Subscriber.ts:190Subscriber._next @ Subscriber.ts:135Subscriber.next @ Subscriber.ts:95Subject.next @ Subject.ts:61EventEmitter.emit @ core.umd.js:6142onError @ core.umd.js:6366onHandleError @ core.umd.js:6241ZoneDelegate.handleError @ zone.js:207Zone.runGuarded @ zone.js:113_loop_1 @ zone.js:379drainMicroTaskQueue @ zone.js:386ZoneTask.invoke @ zone.js:308
core.umd.js:3432 ORIGINAL STACKTRACE:ErrorHandler.handleError @ core.umd.js:3432next @ core.umd.js:6901schedulerFn @ core.umd.js:6150SafeSubscriber.__tryOrUnsub @ Subscriber.ts:238SafeSubscriber.next @ Subscriber.ts:190Subscriber._next @ Subscriber.ts:135Subscriber.next @ Subscriber.ts:95Subject.next @ Subject.ts:61EventEmitter.emit @ core.umd.js:6142onError @ core.umd.js:6366onHandleError @ core.umd.js:6241ZoneDelegate.handleError @ zone.js:207Zone.runGuarded @ zone.js:113_loop_1 @ zone.js:379drainMicroTaskQueue @ zone.js:386ZoneTask.invoke @ zone.js:308
core.umd.js:3433 Error: Uncaught (in promise): undefined
    at resolvePromise (zone.js:429)
    at MdSidenav.eval [as _openPromiseReject] (zone.js:406)
    at MdSidenav._onTransitionEnd (sidenav.js:164)
    at DebugAppView._View_AppComponent0._handle_transitionend_2_2 (AppComponent.ngfactory.js:732)
    at eval (core.umd.js:9678)
    at eval (platform-browser.umd.js:1813)
    at eval (platform-browser.umd.js:1926)
    at ZoneDelegate.invoke (zone.js:203)
    at Object.onInvoke (core.umd.js:6220)
    at ZoneDelegate.invoke (zone.js:202)ErrorHandler.handleError @ core.umd.js:3433next @ core.umd.js:6901schedulerFn @ core.umd.js:6150SafeSubscriber.__tryOrUnsub @ Subscriber.ts:238SafeSubscriber.next @ Subscriber.ts:190Subscriber._next @ Subscriber.ts:135Subscriber.next @ Subscriber.ts:95Subject.next @ Subject.ts:61EventEmitter.emit @ core.umd.js:6142onError @ core.umd.js:6366onHandleError @ core.umd.js:6241ZoneDelegate.handleError @ zone.js:207Zone.runGuarded @ zone.js:113_loop_1 @ zone.js:379drainMicroTaskQueue @ zone.js:386ZoneTask.invoke @ zone.js:308
zone.js:357 Error: Uncaught (in promise): undefined(…)
```

--- a/src/demo-app/sidenav/sidenav-demo.html
+++ b/src/demo-app/sidenav/sidenav-demo.html
@@ -1,3 +1,5 @@
+<h2>Basic Use Case</h2>
+
 <md-sidenav-layout class="demo-sidenav-layout">
   <md-sidenav #start (open)="myinput.focus()" mode="side">
     Start Side Drawer
@@ -34,4 +36,14 @@
   </div>
 </md-sidenav-layout>
 
-<h2>Content after Sidenav</h2>
+<h2>Sidenav Already Opened</h2>
+
+<md-sidenav-layout class="demo-sidenav-layout">
+  <md-sidenav #start2 opened mode="side">
+    Drawer
+  </md-sidenav>
+
+  <div class="demo-sidenav-content">
+    <button md-button (click)="start2.toggle()">Toggle Start Side Drawer</button>
+  </div>
+</md-sidenav-layout>

--- a/src/lib/sidenav/sidenav.spec.ts
+++ b/src/lib/sidenav/sidenav.spec.ts
@@ -221,9 +221,12 @@ describe('MdSidenav', () => {
       endSidenavTransition(fixture);
 
       let sidenavEl = fixture.debugElement.query(By.css('md-sidenav')).nativeElement;
+      let testComponent = fixture.debugElement.query(By.css('md-sidenav')).componentInstance;
 
       expect(sidenavEl.classList).not.toContain('md-sidenav-closed');
       expect(sidenavEl.classList).toContain('md-sidenav-opened');
+
+      expect((testComponent as any)._openPromise).toBeNull();
     });
 
     it('should remove align attr from DOM', () => {

--- a/src/lib/sidenav/sidenav.ts
+++ b/src/lib/sidenav/sidenav.ts
@@ -45,7 +45,7 @@ export class MdDuplicatedSidenavError extends MdError {
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
 })
-export class MdSidenav {
+export class MdSidenav implements AfterContentInit {
   /** Alignment of the sidenav (direction neutral); whether 'start' or 'end'. */
   @Input() align: 'start' | 'end' = 'start';
 
@@ -72,6 +72,15 @@ export class MdSidenav {
    *     If not available we do not hook on transitions.
    */
   constructor(private _elementRef: ElementRef) {}
+
+  ngAfterContentInit() {
+    // This can happen when the sidenav is set to opened in the template and the transition
+    // isn't ended.
+    if (this._openPromise) {
+      this._openPromiseResolve();
+      this._openPromise = null;
+    }
+  }
 
   /**
    * Whether the sidenav is opened. We overload this because we trigger an event when it

